rbi/annotations/railties.rbi

```diff
@@ -42,8 +42,16 @@ class Rails::Application < ::Rails::Engine
 end
 
 class Rails::Engine < ::Rails::Railtie
-  sig { params(block: T.untyped).returns(ActionDispatch::Routing::RouteSet) }
+  sig { params(block: T.nilable(T.proc.bind(ActionDispatch::Routing::Mapper).void)).returns(ActionDispatch::Routing::RouteSet) }
```

Comment:
Why nilable? Does it make sense to not pass a block to `routes`?

Reply:
It seems that the block is indeed optional: https://github.com/rails/rails/blob/main/railties/lib/rails/engine.rb?ref=writesoftwarewell.com#L546. Without it it behaves like an accessor to `@routes`

Reply:
TIL